ZividOpenCV/CaptureUndistortRGB/CaptureUndistortRGB.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+Undistort an RGB image from a ZDF point cloud using Zivid camera intrinsics.
+*/
+
+#include <Zivid/CloudVisualizer.h>
+#include <Zivid/Zivid.h>
+
+#include <opencv2/calib3d/calib3d.hpp>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <iostream>
+
+void pointCloudToRGB(const Zivid::PointCloud &, cv::Mat &);
+void getCameraIntrinsics(const Zivid::Camera &, cv::Mat &, cv::Mat &);
+void displayRGB(const cv::Mat &, const std::string);
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        std::cout << "Connecting to the camera" << std::endl;
+        auto camera = zivid.connectCamera();
+
+        std::cout << "Configuring the camera settings" << std::endl;
+        camera << Zivid::Settings::Iris{ 21 } << Zivid::Settings::ExposureTime{ std::chrono::microseconds{ 20000 } }
+               << Zivid::Settings::Gain{ 1 } << Zivid::Settings::Brightness{ 1.0 };
+
+        std::cout << "Capturing a frame" << std::endl;
+        const auto frame = camera.capture();
+
+        const auto intrinsics = camera.intrinsics();
+
+        std::cout << "Setting up visualization" << std::endl;
+        Zivid::CloudVisualizer vis;
+        zivid.setDefaultComputeDevice(vis.computeDevice());
+
+        std::cout << "Displaying the point cloud" << std::endl;
+        vis.showMaximized();
+        vis.show(frame);
+        vis.resetToFit();
+
+        std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
+        vis.run();
+
+        std::cout << "Converting ZDF point cloud to OpenCV format" << std::endl;
+
+        const auto pointCloud = frame.getPointCloud();
+        cv::Mat rgb((int)pointCloud.height(), (int)pointCloud.width(), CV_8UC3, cv::Scalar(0, 0, 0));
+        pointCloudToRGB(pointCloud, rgb);
+
+        std::cout << "Undistorting the RGB image" << std::endl;
+
+        cv::Size size = rgb.size();
+
+        cv::Mat rgbUndistorted;
+        cv::Mat rgbUndistortedFull;
+
+        cv::Mat distortionCoefficients(cv::Size(1, 5), CV_64FC1, cv::Scalar(0));
+        cv::Mat cameraMatrix(cv::Size(3, 3), CV_64FC1, cv::Scalar(0));
+        cv::Mat optimalCameraMatrix;
```

Comment:
This is now first default initialized on this line, and then re-initialized a few lines below. You can do this on one single line instead, and avoiding un-used temporary objects. That will make the code clearer, and reduce the risk of someone using an improperly initialized variable. It also allows you to make it `const`.

C++ Core Guidelines: [ES.22: Don't declare a variable until you have a value to initialize it with](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es22-dont-declare-a-variable-until-you-have-a-value-to-initialize-it-with)

Reply:
I think this is done if I understand correctly what the problem is.